src/org/thoughtcrime/securesms/CreateProfileActivity.java

```diff
@@ -170,49 +174,38 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
       case Crop.REQUEST_CROP:
         if (resultCode == Activity.RESULT_OK) {
-
-          new AsyncTask<Void, Void, byte[]>() {
-            @Override
-            protected byte[] doInBackground(Void... params) {
-              try {
-                try {
-                  Uri imageUri = Crop.getOutput(data);
-                  Bitmap bitmap = MediaStore.Images.Media.getBitmap(getContentResolver(), imageUri);
-                  ByteArrayOutputStream stream = new ByteArrayOutputStream();
-                  bitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream);
-                  return stream.toByteArray();
-                } catch (Exception any) {
-                  Log.e(TAG, "could not send raw JPG to core. Using scaled JPG.", any);
-                }
-                BitmapUtil.ScaleResult result =
-                    BitmapUtil.createScaledBytes(CreateProfileActivity.this, Crop.getOutput(data), new ProfileMediaConstraints());
-                return result.getBitmap();
-              } catch (BitmapDecodingException e) {
-                Log.w(TAG, e);
-                return null;
-              }
-            }
-
-            @Override
-            protected void onPostExecute(byte[] result) {
-              if (result != null) {
-                avatarBytes = result;
-                GlideApp.with(CreateProfileActivity.this)
-                        .load(avatarBytes)
-                        .skipMemoryCache(true)
-                        .diskCacheStrategy(DiskCacheStrategy.NONE)
-                        .circleCrop()
-                        .into(avatar);
-              } else {
-                Toast.makeText(CreateProfileActivity.this, R.string.error, Toast.LENGTH_LONG).show();
-              }
-            }
-          }.execute();
+          setAvatarView(data);
         }
         break;
     }
   }
 
+  private void setAvatarView(Intent data) {
+    final Uri output = Crop.getOutput(data);
+    final ProfileMediaConstraints constraints = new ProfileMediaConstraints();
+    GlideApp.with(this)
+            .asBitmap()
+            .load(output)
+            .skipMemoryCache(true)
+            .diskCacheStrategy(DiskCacheStrategy.NONE)
+            .centerCrop()
+            .override(constraints.getImageMaxWidth(this), constraints.getImageMaxHeight(this))
+        .into(new SimpleTarget<Bitmap>() {
```

Comment:
Just a note: SimpleTarget is deprecated and "unsafe", but since it's also used in GroupCreationActivity, we can just leave it for now I think, and then mabye one day we get to fix all the deprecation warnings. Or maybe not, as far as we don't get any problems with it.